### DISCUSSION TITLE
multi tr elements inside template does not work

### DISCUSSION THREAD

**Discussion by jrtozza** [alpinejs/alpine#3427] [1. Help] [answered]
<table  x-data="{list:{un:{alpha:'uno', beta:'dos', show:true}, deux:{alpha:'tres', beta:'quatro', show:false}}}" >
				<tbody>
					<template x-for="(val,k) in list">
						<tr>
							<td x-text="val.alpha"></td>
							<td @click="val.show=true; console.log(list)">Editer</td>						
						</tr>
						
						<tr >
							<template x-if="val.show">
								<td colspan=2  x-text="val.beta"></td>							
							</template>
						</tr>
						
							
						<tr >							
								<td colspan=2  x-show="val.show">oooo</td>							
						</tr>						
						
					</template>						
				</tbody>
			</table>

Only the first TR est displayed, if I try the same code with a div child for the x-for template, like below, it works. but I have to change for div structure.

<div  x-data="{selected:0, list:{un:{alpha:'uno', beta:'dos', show:true}, deux:{alpha:'tres', beta:'quatro', show:false}}}" >
				
					<template x-for="(val,k) in list">
						<div>
						
						
							<div>
								<div x-text="val.alpha"></div>
								<div @click="selected=k; console.log(list); console.log(selected)">Editer</div>						
							</div>
							
							<div>
								<template x-if="k==selected">
									<div  x-text="val.beta"></div>							
								</template>
							</div>
							
								
							<div>							
									<div  x-show="k==selected">oooo</div>							
							</div>	
							
												
						</div>	
					</template>						
			
			</div>

So, template seems to accept only one child. The problem is that I want it with TR  and put a DIV around the tr elements does not work. How can I do ?

**Comment by SimoTod** [comment]
You can have multiple tbody in your table

```html
<template ...>
  <tbody>
     <tr>...</tr>
     ... 
     <tr>...</tr>
  </tbody>
</template>
```

**Comment by ekwoka** [comment, marked as answer]
Alternatively to the above you can give the wrapper div `display: contents` (`contents` in TW) which makes it a "passthrough" element for layout purposes. Not unlike a fragment in react

**Reply by K4zoku** [reply]
Thank you so much
